platform/ios/src/MGLMapboxEvents.m

```diff
@@ -543,7 +543,17 @@ - (void) postEvents:(NS_ARRAY_OF(MGLMapboxEventAttributes *) *)events {
             [request setHTTPBody:jsonData];
 
             // Send non blocking HTTP Request to server
-            [[_session dataTaskWithRequest:request] resume];
+            if (strongSelf.session) {
+                [[strongSelf.session dataTaskWithRequest:request] resume];
+            } else {
+                // Session is invalidated when telemetry is paused, which means
+                // we have to manually manage its lifecycle here for the turnstile.
+                strongSelf.session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]
+                                                                   delegate:strongSelf
+                                                              delegateQueue:nil];
+                [[strongSelf.session dataTaskWithRequest:request] resume];
+                [strongSelf.session finishTasksAndInvalidate];
```

Comment:
Making this more specific to the turnstile sounds good, forget why I didn't go with `isPaused` here but will look at casing this to the turnstile only.

:+1: on using a standalone `NSURLSession` here.